core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2342,6 +2342,12 @@ private[spark] object Utils extends Logging {
    * Return the initial number of executors for dynamic allocation.
    */
   def getDynamicAllocationInitialExecutors(conf: SparkConf): Int = {
+    if (conf.get(DYN_ALLOCATION_INITIAL_EXECUTORS) < conf.get(DYN_ALLOCATION_MIN_EXECUTORS)) {
+      logWarning(s"${DYN_ALLOCATION_INITIAL_EXECUTORS.key} less than " +
+        s"${DYN_ALLOCATION_MIN_EXECUTORS.key} is invalid, will use " +
+          s"${DYN_ALLOCATION_MIN_EXECUTORS} instead.")
```

Comment:
executor instances is what gets set if user puts in --num-executors which I see easily happening now and if min is already set and > then they are different and we will use min so we should warn here also.